src/mail_room.py

```diff
@@ -0,0 +1,166 @@
+            main_controller()
+        else:
+            print_report(to_print)
+            write_to_file(to_print)
+            main_controller()
+
+
+def prompt_usr(prompt):
+    """Return the user response to a prompt."""
+    try:
+        input = raw_input
+    except NameError:
+        pass
+    usr_answer = input('{} \n>>>>>> '.format(prompt))
+    return usr_answer.lower()
+
+
+def validator(usr_response):
+    """Check the user answer."""
+    usr_response = usr_response.lower()
+    if usr_response == "send a thank you":
+        return True
+    elif usr_response == "create a report":
+        return False
+    elif usr_response == "q":
+        sys.exit()
+    else:
+        return "error"
+
+
+def send_or_report():
+    """Send a thank you or create a report."""
+    usr_response = prompt_usr(prompts[0])
+    validate = validator(usr_response)
+    if validate is True:
+        return True
+    elif validate is False:
+        return False
+    elif validate == "error":
+        print("Please choose a valid option")
+        send_or_report()
+
+
+def get_name():
+    """Get gifter name from user."""
+    while True:
+        usr_response = prompt_usr(prompts[1])
+        usr_response = usr_response.lower()
+        if usr_response == "q":
+            sys.exit()
+        elif usr_response == "l":
+            name_list = create_list(GIFTER_DICT)
+            print(name_list)
+        else:
+            break
+    return usr_response
+
+
+def check_dict(name, donate_dict):
+    """Check to see if name exists in dictionary."""
+    if name in donate_dict:
+        return True
+    else:
+        donate_dict.update({str(name): {'name': str(name), 'total_ammt': 0, 'most_recent_ammt': 0, 'times_given': 0}})
+        return False
+
+
+def get_ammt():
+    """Get donation ammount from user."""
+    while True:
+        try:
+            usr_response = prompt_usr(prompts[2])
+            usr_response = usr_response.lower()
+            if usr_response == "q":
+                sys.exit()
+            return int(usr_response)
+        except ValueError:
+            print("Please type a full dollar ammount with no special characters")
+
+
+def update_dict(name, ammt):
+    """Update values for doner in dict."""
+    temp_key = GIFTER_DICT.get(name)
+    get_total = temp_key.get('total_ammt')
+    total = int(get_total) + ammt
+    get_total_times = temp_key.get('times_given')
+    total_times = (get_total_times + 1)
+    GIFTER_DICT.update({name: {'name': name, 'total_ammt': total, 'most_recent_ammt': ammt, 'times_given': total_times}})
+
+
+def send_email(name, ammt):
+    """Print out customized email text."""
+    print(email_text.format(name, ammt))
+    return email_text.format(name, ammt)
+
+
+def create_list(donate_dict):
+    """Create a list of names that have given before."""
+    if len(donate_dict) > 0:
+        name_list = list(donate_dict.keys())
+        return name_list
+    else:
+        print("There are no names")
```

Comment:
Why print anything here.  The stated purpose of this function is to create a list of names by extracting the keys from a dict.  If the number of entries in the dict is 0, then what is name_list when you call `list(dict.keys())`?  It's an empty list.  This is a "falsy" value, and you can allow the calling context where this function is used decide whether to report that to the user or not.  There is no reason to do so here.